python/dgl/distributed/optim/pytorch/sparse_optim.py

```diff
@@ -85,36 +284,67 @@ def step(self):
                         grad_i = grads[mask]
 
                         if trainers_per_server <= 1:
-                            idx_split_size.append(th.tensor([idx_i.shape[0]], dtype=th.int64))
+                            idx_split_size.append(
+                                th.tensor([idx_i.shape[0]], dtype=th.int64)
+                            )
                             idics_list.append(idx_i)
                             grad_list.append(grad_i)
                         else:
-                            kv_idx_split = th.remainder(idx_i, trainers_per_server).long()
+                            kv_idx_split = th.remainder(
+                                idx_i, trainers_per_server
+                            ).long()
                             for j in range(trainers_per_server):
                                 mask = kv_idx_split == j
                                 idx_j = idx_i[mask]
                                 grad_j = grad_i[mask]
-                                idx_split_size.append(th.tensor([idx_j.shape[0]], dtype=th.int64))
+                                idx_split_size.append(
+                                    th.tensor([idx_j.shape[0]], dtype=th.int64)
+                                )
                                 idics_list.append(idx_j)
                                 grad_list.append(grad_j)
 
                     # if one machine launch multiple KVServer, they share the same storage.
-                    # For each machine, the pytorch rank is num_trainers * machine_id + i
+                    # For each machine, the pytorch rank is num_trainers *
+                    # machine_id + i
 
                     # use scatter to sync across trainers about the p2p tensor size
                     # Note: If we have GPU nccl support, we can use all_to_all to
                     # sync information here
-                    gather_list = list(th.empty([self._world_size],
-                                                dtype=th.int64).chunk(self._world_size))
-                    alltoall_cpu(self._rank, self._world_size, gather_list, idx_split_size)
+                    gather_list = list(
+                        th.empty([self._world_size], dtype=th.int64).chunk(
+                            self._world_size
+                        )
+                    )
```

Comment:
Is there any particular reason to construct a list of tensors in this way instead of
```python
[th.empty([1], dtype=th.int64) for _ in range(self._world_size)]
```
Also, does it have to be wrapped as a list?

Reply:
It is old code and changed by formatting tools,  not sure why it written in this way.

Reply:
Fair enough.